include/exadg/incompressible_navier_stokes/postprocessor/line_plot_calculation.cpp

```diff
@@ -64,45 +89,43 @@ LinePlotCalculator<dim, Number>::evaluate(VectorType const & velocity,
     // precision
     unsigned int const precision = data.line_data.precision;
 
-    // loop over all lines
-    for(typename std::vector<std::shared_ptr<Line<dim>>>::const_iterator line =
-          data.line_data.lines.begin();
-        line != data.line_data.lines.end();
-        ++line)
+    LinearAlgebra::distributed::Vector<double> ghosted_pressure(pressure.get_partitioner());
+    ghosted_pressure.copy_locally_owned_data_from(pressure);
+    ghosted_pressure.update_ghost_values();
+
+    std::vector<double> pressure_values =
+      VectorTools::point_values<1>(*evaluation_cache, *dof_handler_pressure, ghosted_pressure);
+
+    LinearAlgebra::distributed::Vector<double> ghosted_velocity(velocity.get_partitioner());
+    ghosted_velocity.copy_locally_owned_data_from(velocity);
+    ghosted_velocity.update_ghost_values();
+
+    std::vector<Tensor<1, dim>> velocity_values =
+      VectorTools::point_values<dim>(*evaluation_cache, *dof_handler_velocity, ghosted_velocity);
+
+    // write output to file on rank 0
+    if(Utilities::MPI::this_mpi_process(mpi_comm) == 0)
     {
-      // store all points along current line in a vector
-      unsigned int            n_points = (*line)->n_points;
-      std::vector<Point<dim>> points(n_points);
-
-      // we consider straight lines with an equidistant distribution of points along the line
-      for(unsigned int i = 0; i < n_points; ++i)
-        points[i] =
-          (*line)->begin + double(i) / double(n_points - 1) * ((*line)->end - (*line)->begin);
-
-      // filename prefix for current line
-      std::string filename_prefix = data.line_data.directory + (*line)->name;
-
-      // write output for all specified quantities
-      for(std::vector<std::shared_ptr<Quantity>>::const_iterator quantity =
-            (*line)->quantities.begin();
-          quantity != (*line)->quantities.end();
-          ++quantity)
+      std::size_t point_offset = 0;
+      // evaluate by looping over all lines
+      for(typename std::vector<std::shared_ptr<Line<dim>>>::const_iterator line =
+            data.line_data.lines.begin();
+          line != data.line_data.lines.end();
+          ++line)
       {
-        if((*quantity)->type == QuantityType::Velocity)
-        {
-          std::vector<Tensor<1, dim, Number>> solution_vector(n_points);
+        // store all points along current line in a vector
```

Comment:
this comment seems to be wrong now, and I currently also do not see where you actually use the variable ``n_points``.